File: CVBuilder.Core/Models/TemplateModel.cs
namespace CVBuilder.Core.Models
{
    public class TemplateModel
    {
        public int Id { get; set; }

        public string Name { get; set; } = default!;
        public string? Description { get; set; }
        public string? PreviewImageUrl { get; set; }

        public string CssContent { get; set; } = ""; 

        public bool IsActive { get; set; } = true;
    }
}

File: CVBuilder.Web/src/app/models/template.model.ts
export interface Template {
  id: number;

  // Display & Identity
  name: string;
  description?: string | null;
  previewImageUrl: string;
  templateKey: string;

  // Rendering Configuration
  htmlTemplatePath: string;
  cssClass: string;
  fontFamily: string;
  primaryColor: string;
  accentColor?: string | null;

  // Layout Options
  isTwoColumn: boolean;
  paperSize: string;

  // Usage Flags
  isDefault: boolean;
  isActive: boolean;

  // Optional metadata
  category: string;
  notes?: string | null;
}
